Source/Core/Core/HW/SI/SI.cpp

```diff
@@ -368,8 +369,35 @@ void DoState(PointerWrap& p)
   p.Do(s_si_buffer);
 }
 
+static void DeviceEventCallback(u64 userdata, s64 cyclesLate)
+{
+  s_channel[userdata].device->OnEvent(cyclesLate);
+}
+
+static void RegisterEvents()
+{
+  s_change_device_event = CoreTiming::RegisterEvent("ChangeSIDevice", ChangeDeviceCallback);
+  s_tranfer_pending_event = CoreTiming::RegisterEvent("SITransferPending", RunSIBuffer);
+
+  for (int i = 0; i < MAX_SI_CHANNELS; ++i)
+    s_device_events[i] =
+        CoreTiming::RegisterEvent(fmt::format("SIEventChannel{}", i), DeviceEventCallback);
```

Comment:
Since this spans multiple lines, please add some curlies:
```suggestion
  for (int i = 0; i < MAX_SI_CHANNELS; ++i)
  {
    s_device_events[i] =
        CoreTiming::RegisterEvent(fmt::format("SIEventChannel{}", i), DeviceEventCallback);
  }
```